scalyr_agent/builtin_monitors/syslog_utils.py

```diff
@@ -0,0 +1,211 @@
+from __future__ import absolute_import
+
+import glob
+import os
+import re
+import copy
+import time
+from string import Template
+
+import six
+
+from scalyr_agent.json_lib import JsonObject
+
+from scalyr_agent.builtin_monitors.journald_utils import LogConfigManager
+
+import scalyr_agent.scalyr_logging as scalyr_logging
+from six.moves import range
+
+global_log = scalyr_logging.getLogger(__name__)
+
+
+class SyslogLogConfigManager(LogConfigManager):
+    def __init__(
+        self,
+        global_config,
+        formatter,
+        max_log_size=20 * 1024 * 1024,
+        max_log_rotations=2,
+        extra_config=None,
+    ):
+        LogConfigManager.__init__(
+            self,
+            global_config,
+            formatter,
+            max_log_size=max_log_size,
+            max_log_rotations=max_log_rotations,
+            extra_config=extra_config,
+        )
+        self.current_log_files = []
+        self.log_deleter = LogDeleter(
+            extra_config.get("docker_check_for_unused_logs_mins", 60),
```

Comment:
Do we intentionally use keys starting with ``docker_`` here?

I assume we want to use the same additional options for docker and syslog monitor, or?

Reply:
Yes this is intentional, these are fields that are only useful in cases where the syslog monitor is running on docker mode, and the log deleter only does anything in docker mode as well.